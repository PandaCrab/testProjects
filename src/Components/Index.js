import { ShippingInfo } from './Forms/ShippingInfo';
import { BillingInfo } from './Forms/BillingInfo';
import { Payment } from './Forms/Payment';
import { OrderPlate } from './OrderPlate/OrderPlate';


export { ShippingInfo, BillingInfo, Payment, OrderPlate };
