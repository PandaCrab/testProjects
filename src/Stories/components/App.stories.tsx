import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ApolloProvider } from '@apollo/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '../../redux/store';
import App from '../../App';
import { apolloClient } from '../..';

export default {
    title: 'App',
    component: App
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = () => (
    <ApolloProvider client={apolloClient}>
        <Provider store={store}>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </Provider>
    </ApolloProvider>
);

export const Header = Template.bind({});
