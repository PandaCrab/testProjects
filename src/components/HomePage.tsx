import React from 'react';
import { useNavigate } from 'react-router-dom';

import { 
    StartPage,
    NavigationButton
} from '../styles/HomePageStyles';

const HomePage = () => {
    const navigate = useNavigate();

    return (
        <>
            <StartPage>
                <h1>Hello World</h1>
                <NavigationButton onClick={() => navigate('/shopping')}>Start shopping</NavigationButton>
            </StartPage>
        </>
    );
};

export default HomePage;
