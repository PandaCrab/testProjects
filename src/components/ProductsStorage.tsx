import { useState } from 'react';
import { 
    SortingState,
    IntegratedSorting,
    RowDetailState
} from '@devexpress/dx-react-grid';
import {
  Grid,
  Table,
  TableHeaderRow,
  TableRowDetail,
} from '@devexpress/dx-react-grid-material-ui';
// import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useQuery } from '@apollo/client';

// import { AppDispatch, RootState } from '../types';

// import { getProductsStorage } from '../redux/ducks/stuff';
import { Heading, NavigationButton, StartPage } from '../styles/HomePageStyles';
import { TAKE_PRODUCTS_FROM_STORAGE } from '../api';

interface graphQlData {
    id: string,
    name: string,
    price: number,
    quantity: number
}

const RowDetail = ({ row }: any) => (
    <div>
        {row.quantity > 0 ? 
        `This item can be send today` :
        `Check later, out of stock`
        }
    </div>
)

const ProductsStorage = () => {
    const { error, data } = useQuery(TAKE_PRODUCTS_FROM_STORAGE);
    const navigate = useNavigate();
    // const dispatch = useDispatch<AppDispatch>();
    // const productsStorage = useSelector((state: RootState) => state.order.productsStorage);

    // useEffect(() => { 
    //     dispatch(getProductsStorage())
    // // eslint-disable-next-line react-hooks/exhaustive-deps
    // }, []);

    const [columns] = useState([
        { name: 'id', title: 'Id' },
        { name: 'name', title: 'Name' },
        { name: 'price', title: 'Price' },
        { name: 'quantity', title: 'Quantity' }
    ]);

    const [tableColumnExtensions] = useState([
        { columnName: 'id', width: '10%' },
        { columnName: 'name', width: '50%' },
        { columnName: 'price', width: '10%' },
        { columnName: 'quantity', width: '15%' },
      ]);

    return (
        <StartPage>
            <Heading>
                <h1>Hello in storage</h1>  <NavigationButton onClick={() => navigate('/')} >Home</NavigationButton>
            </Heading>
            <Grid
                rows={data.productsStorage.map(({ id, name, price, quantity }: graphQlData) => ({
                    id,
                    name,
                    price,
                    quantity
                }))}
                columns={columns}
            >
                <SortingState
                    defaultSorting={[{ columnName: 'name', direction: 'asc' }]}
                />
                <IntegratedSorting />
                <RowDetailState />

                <Table columnExtensions={tableColumnExtensions} />
                <TableHeaderRow showSortingControls />
                <TableRowDetail 
                    contentComponent={RowDetail}
                />
            </Grid>
        </StartPage>
    );
};

export default ProductsStorage;
