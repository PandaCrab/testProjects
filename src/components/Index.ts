import ShippingInfo from './forms/ShippingInfo';
import BillingInfo from './forms/BillingInfo';
import Payment from './forms/Payment';
import OrderPlate from './orderPlate/OrderPlate';
import SuccessBuying from './forms/SuccessBuying';
import Navigate from '../Navigate';
import Stuff from './orderPlate/Stuff';
import StuffItems from './orderPlate/StuffItems';
import Loader from './Loader';
import CardToPrint from './forms/CardToPrint';
import DropdownAddresses from './forms/dropdownAddresses';
import CountriesSelect from './forms/CountriesSelect';
import HomePage from './HomePage';
import ProductsStorage from './ProductsStorage';
import SaleChart from './SaleChart';

export { ShippingInfo,
    BillingInfo, 
    Payment, 
    OrderPlate, 
    SuccessBuying, 
    Navigate,
    Stuff,
    StuffItems,
    Loader,
    CardToPrint,
    DropdownAddresses,
    CountriesSelect,
    HomePage,
    ProductsStorage,
    SaleChart
};
