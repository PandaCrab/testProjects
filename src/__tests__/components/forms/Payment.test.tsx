import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { Payment } from '../../../components/Index';

const initialState = {}
const mockStore = configureStore();
let store;

describe('Payment component:', () => {
    
    it('should render correctly Order plate component', () => {
        store = mockStore(initialState);
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <Payment />
                </BrowserRouter>
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});

describe('Check cardholder field in Payment component', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <Payment />
            </BrowserRouter>
        </Provider>)

    const cardholderInput = component.find('input[name="cardholder"]')

    it('should have carholder field', () => {
        expect(cardholderInput.length).toEqual(1);
    });

    it('should set the name value on change event', () => {
        cardholderInput.simulate('change', {
            target: { value: 'Pupkin Vasya' } 
        });
        
        const inputAfterChange = component.find('input[name="cardholder"]');
        expect(inputAfterChange.prop('value')).toEqual('Pupkin Vasya');
    });
});

describe('Check card number field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <Payment />
            </BrowserRouter>
        </Provider>)

    const cardNumInput = component.find('input[name="cardNum"]')

    it('should have carholder field', () => {
        expect(cardNumInput.length).toEqual(1);
    });

    it('should set the name value on change event', () => {
        cardNumInput.simulate('change', {
            target: { value: '4334 5533 5533 3344' } 
        });
        
        const inputAfterChange = component.find('input[name="cardNum"]');
        expect(inputAfterChange.prop('value')).toEqual('4334 5533 5533 3344');
    });
});

describe('Check date field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <Payment />
            </BrowserRouter>
        </Provider>)

    const dateInput = component.find('input[name="date"]')

    it('should have carholder field', () => {
        expect(dateInput.length).toEqual(1);
    });

    it('should set the name value on change event', () => {
        dateInput.simulate('change', {
            target: { value: '12/33' } 
        });
        
        const inputAfterChange = component.find('input[name="date"]');
        expect(inputAfterChange.prop('value')).toEqual('12/33');
    });
});

describe('Check code field', () => {
    store = mockStore(initialState)
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <Payment />
            </BrowserRouter>
        </Provider>)

    const cvvInput = component.find('input[name="cvv"]')

    it('should have carholder field', () => {
        expect(cvvInput.length).toEqual(1);
    });

    it('should set the name value on change event', () => {
        cvvInput.simulate('change', {
            target: { value: '321' } 
        });
        
        const inputAfterChange = component.find('input[name="cvv"]');
        expect(inputAfterChange.prop('value')).toEqual('321');
    });
});

    describe('Submit button', () => {
    store = mockStore(initialState);
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <Payment />
            </BrowserRouter>
        </Provider>
    );

    it('should have submit button', () => {
        expect(component.find('button[type="submit"]').length).toEqual(1);
    });

    it('should submit on click', () => {
        const mockHandleSubmit = jest.fn();

        component.find('form').simulate('submit', mockHandleSubmit());

        expect(mockHandleSubmit).toHaveBeenCalledTimes(1);
    });
});
