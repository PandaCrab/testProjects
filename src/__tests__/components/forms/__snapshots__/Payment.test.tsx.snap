// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment component: should render correctly Order plate component 1`] = `
Array [
  <section
    className="sc-BeQoi dbxsDW"
  >
    <a
      href="/shipping"
      onClick={[Function]}
    >
      Shipping
    </a>
     &gt;
    <a
      href="/billing"
      onClick={[Function]}
    >
      Billing
    </a>
     &gt;
    <a
      href="/payment"
      onClick={[Function]}
    >
      Payment
    </a>
  </section>,
  <div
    className="sc-bczRLJ iLjvJl"
  >
    <section
      className="sc-kDDrLX hFYHf"
    >
      <h4
        className="sc-iqcoie bSFIcJ"
      >
        Payment
      </h4>
    </section>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="mb-4 position-relative"
      >
        <label
          className="mb-0 form-label"
        >
          cardholder Name
        </label>
        <input
          className="form-control"
          id="cardholder"
          name="cardholder"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Name as it appears on your card"
          type="text"
          value=""
        />
        <div
          className="invalid-tooltip"
        />
      </div>
      <div
        className="mb-4 position-relative"
      >
        <label
          className="mb-1 form-label"
        >
          Card Number
        </label>
        <div
          className="input-group"
        >
          <input
            aria-label="Card number"
            autoComplete="cc-number"
            className="sc-iBkjds bHhSWR form-control"
            id="cardNum"
            name="cardNum"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="XXXX XXXX XXXX XXXX XXXX"
            type="text"
            value=""
          />
          <div
            className="invalid-tooltip"
          />
        </div>
      </div>
      <div
        className="mb-4 position-relative"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-4 col-5"
          >
            <label
              className="mb-1 form-label"
            >
              Expire Date
            </label>
            <input
              aria-label="Expiry date in format MM YY"
              autoComplete="cc-exp"
              className="form-control"
              id="date"
              name="date"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              placeholder="MM/YY"
              type="text"
              value=""
            />
            <div
              className="invalid-tooltip"
            />
          </div>
          <div
            className="col-sm-5 col-6"
          >
            <label
              className="mb-1 form-label"
            >
              Security code
            </label>
            <input
              aria-label="CVC"
              autoComplete="cc-csc"
              className="form-control"
              id="cvv"
              name="cvv"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              placeholder="CVC"
              type="text"
              value=""
            />
            <div
              className="invalid-tooltip"
            />
          </div>
        </div>
      </div>
      <button
        className="sc-jqUVSM lirFSS btn btn-primary"
        disabled={false}
        type="submit"
      >
        Pay Securely
      </button>
    </form>
  </div>,
]
`;
